pkg/engine2/path_selection/path_expansion.go

```diff
@@ -128,23 +135,45 @@ func renameAndReplaceInTempGraph(
 				}
 				id.Name = fmt.Sprintf("%s-%d", name, suffix)
 			}
-			_, props, err := input.TempGraph.VertexWithProperties(oldId)
-			if err == nil && props.Attributes != nil {
+			if props.Attributes != nil {
 				props.Attributes["new_id"] = id.String()
 			}
+			phantomRes := r
+			r, err = knowledgebase.CreateResource(ctx.KnowledgeBase(), id)
+			if err != nil {
+				errs = errors.Join(errs, err)
+				continue
+			}
+			r.Properties = phantomRes.Properties
+			phantomRes.Properties = make(construct.Properties)
+		}
+		err = g.AddVertex(r)
+		switch {
+		case errors.Is(err, graph.ErrVertexAlreadyExists):
+			r, err = g.Vertex(id)
+			if err != nil {
+				errs = errors.Join(errs, err)
+				continue
+			}
+
+		case err != nil:
+			errs = errors.Join(errs, err)
+			continue
+		}
+		result[i] = r
+		if i > 0 {
+			err = g.AddEdge(result[i-1].ID, r.ID)
```

Comment:
Not sure why my change broke the k8s functional test, but this change (`id` -> `r.ID`) fixed it.